### PR TITLE
Add HLint

### PR DESCRIPTION
**Pull request by blackheaven**
I'm trying to tackle #63

### CODE REVIEW

File: .github/workflows/hlint.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: lint
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - 'releases/*'
```
Comment (chshersh) [inline .github/workflows/hlint.yml:7]: Let's move this job inside the `ci.yml` file. It already has proper `on` criteria:

https://github.com/chshersh/iris/blob/1bede5f2c7211e633c6356f03b1dc5f201326717/.github/workflows/ci.yml#L3-L10

File: .github/workflows/hlint.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    - name: 'Set up HLint'
+      uses: rwe/actions-hlint-setup@v1
+      with:
+        version: '3.1.6'
+
+    - name: 'Run HLint'
+      uses: rwe/actions-hlint-run@v2
+      with:
+        path: '["src/", "test/"]'
+        fail-on: warning
```
Comment (chshersh) [inline .github/workflows/hlint.yml:24]: I haven't used `rwe/actions-hlint-setup` and `rwe/actions-hlint-run` actions before.

I would prefer to use official Haskell actions under the `haskell` github org:

* https://github.com/haskell/actions

But even then, I'm not sure what is the difference between those two actions and a simple job that just downloads the `hlint` executable and runs it. I used to configure `hlint` before with this simple setup:

* https://github.com/kowainik/stan/blob/dcbc8682a964ae6f642ae37adbd46f6545d39aa6/.github/workflows/ci.yml#L102-L117

**Review by chshersh — COMMENTED** [review]
Thanks for the contribution, @blackheaven!

I have a few questions and suggestions to clarify the proper CI configuration though

File: .github/workflows/hlint.yml (hunk 2 above)
Comment (blackheaven) [reply .github/workflows/hlint.yml]: IIUC:
- `rwe/actions-hlint-setup` deals with the subtleties of the underlying OS (ie. download le `exe` on windows, and on)
- `rwe/actions-hlint-run` only run hlint on the new/changed files

I update the PR according to your feedback, I'm open to new comments/requests.

**Review by chshersh — APPROVED** [review]
@blackheaven Thanks a lot for the contribution! I agree that we can always improve the action further if we find lacking something but for now we can start with the simplest thing that works 🙂

**Pull request merged**